libcatalyst: use git instead of fixed urls

vicentebolea opened the pull request:
I had difficulties custom installing specific versions of Catalyst, here I change URL to git to be able to install commit at wish.

    The issue is that we are not not able to install (Fetch URL error) any
    version of catalyst other than the specified in the spack package.py.
    This very version is accessible only because it is cached by Spack. The
    real URL does not exist anymore, I believe the reason is that there used
    to be a tag in catalyst that does not exist anymore.

mathstuf commented:
> I had difficulties

Such as?

vicentebolea commented:
@mathstuf 

The issue is that I am not able to install (Fetch URL error) any version of catalyst other than the specified in the spack package.py. This very version is accessible only because it is cached by Spack. The real URL does not exist anymore, I believe the reason is that there used to be a tag in catalyst that does not exist anymore.

mathstuf reviewed (dismissed):
It'd be nice if that information was provided in the commit messages. Otherwise it seems fine.

vicentebolea committed:
libcatalyst: use git instead of fixed urls

The issue is that we are not not able to install (Fetch URL error) any
version of catalyst other than the specified in the spack package.py.
This very version is accessible only because it is cached by Spack. The
real URL does not exist anymore, I believe the reason is that there used
to be a tag in catalyst that does not exist anymore.

tldahlgren reviewed (approved):
Confirmed commit exists in the repository and that 'spack stage' of the version succeeds.

tldahlgren reviewed (commented):
Still testing the fetch.

tldahlgren commented:
@vicentebolea What version(s) are you trying to fetch with this change where removing the url helps?

I have not been able to convince myself this works as expected.

vicentebolea commented:
> I have not been able to convince myself this works as expected.

libcatalyst@3f7871c0a2e737cb9ed35fc1c2208456fcc00a0e
